src/main/java/bio/terra/service/GoogleBillingService.java

```diff
@@ -0,0 +1,90 @@
+package bio.terra.service;
+
+import bio.terra.metadata.BillingProfile;
+import bio.terra.service.exception.ExternalServiceException;
+import com.google.api.client.googleapis.auth.oauth2.GoogleCredential;
+import com.google.api.client.googleapis.javanet.GoogleNetHttpTransport;
+import com.google.api.client.googleapis.json.GoogleJsonResponseException;
+import com.google.api.client.http.HttpTransport;
+import com.google.api.client.json.JsonFactory;
+import com.google.api.client.json.jackson2.JacksonFactory;
+import com.google.api.services.cloudbilling.Cloudbilling;
+import com.google.api.services.cloudbilling.model.TestIamPermissionsRequest;
+import org.springframework.context.annotation.Profile;
+import org.springframework.stereotype.Service;
+
+import java.io.IOException;
+import java.security.GeneralSecurityException;
+import java.util.Collections;
+import java.util.List;
+
+@Service
+@Profile("google")
+public class GoogleBillingService implements BillingService {
+
+    private final Cloudbilling cloudbilling;
+
+    public GoogleBillingService() throws IOException, GeneralSecurityException {
+        // Authentication is provided by the 'gcloud' tool when running locally
+        // and by built-in service accounts when running on GAE, GCE, or GKE.
+        GoogleCredential credential = GoogleCredential.getApplicationDefault();
```

Comment:
Curious why you do this in the constructor. Won't the access token in the credential time out? I think it takes some code to automatically refresh it.

Since this is a very rare operation, perhaps just do this as part of the operation and not in the constructor.

Now that I think about it, that might be a general problem in our code.

Reply:
Good point, will do.